xarray/backends/netCDF4_.py

```diff
@@ -210,6 +210,10 @@ def _extract_nc4_variable_encoding(variable, raise_on_invalid=False,
         if chunks_too_big or changed_shape:
             del encoding['chunksizes']
 
+    var_has_unlim_dim = any(dim in unlimited_dims for dim in variable.dims)
+    if var_has_unlim_dim and 'contiguous' in encoding.keys():
```

Comment:
This should also check for `not raise_on_invalid`, like what we do above for `chunksizes`. Otherwise, if a user specified an invalid encoding by hand we would silently ignore their error.

Reply:
@shoyer I think this is what you mean.

Reply:
https://github.com/pydata/xarray/pull/2941/files#diff-3e64cce1e0c1e1cb0c1745c3723c80ffR214

Reply:
Thanks @jmccreight . We prefer using `()` to wrap long lines instead of using a `\`. I've made the fix.

Reply:
Thanks. I was, in fact, very uncertain about the preferred approach.